src/System.Buffers/ref/System.Buffers.cs

```diff
@@ -16,3 +16,111 @@ public abstract class ArrayPool<T>
         public abstract void Return(T[] array, bool clearArray = false);
     }
 }
+
+namespace System.Buffers.Binary
+{
+    [System.CLSCompliantAttribute(false)]
+    public static class SpanBinaryExtensions
+    {
+        public static sbyte Reverse(this sbyte value) { throw null; }
+        public static byte Reverse(this byte value) { throw null; }
+        public static short Reverse(this short value) { throw null; }
+        public static ushort Reverse(this ushort value) { throw null; }
+        public static int Reverse(this int value) { throw null; }
+        public static uint Reverse(this uint value) { throw null; }
+        public static long Reverse(this long value) { throw null; }
+        public static ulong Reverse(this ulong value) { throw null; }
+
+        public static T Read<T>(this Span<byte> buffer) where T : struct { throw null; }
```

Comment:
The `Read<T>` and `Write<T>` APIs should probably prevent callers from using non-blittable `T`, otherwise we risk reading and writing memory addresses to the underlying stream. (Additionally, are there cases where the field offsets can change between runtimes or architectures? I couldn't find any examples.)

Reply:
Regarding this, we are considering adding a Primitive attribute to constrain T to blittable types only (structs without reference types). cc @KrzysztofCwalina, @jaredpar